### PR TITLE
Secure UI API Layer

### LINKED ISSUE #8: Authentication and authorization for GraphQL
Confirm these statements before you submit the issue:

- [ Y] I have searched open and closed issues for duplicates.
- [ Y] I have read the contributing guidelines.
---

**Description**

Introduces authentication and authorization for GraphQL.

**Reasons**

The GraphQL facade caches resources and does not forward every request to the API Server. So, the user token is not validated against RBAC. The consequence is that all authenticated users can perform all operations supported by the GraphQL facade. This is a vulnerability.

To patch this vulnerability, Kyma developers need an additional authorization layer before the GraphQL facade.

**Attachments**

More information about the change has been described in the following ADR: https://github.com/kyma-project/community/blob/master/architecture-decision-records/adr-015-Authorization_for_GraphQL.md

### PR DESCRIPTION
**Pull request by damianpacierpnikatsap**
Confirm these statements before you submit your pull request:

- [y] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [y] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [y] I have tested my changes.
- [y] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:

- Introduces authentication and authorization for GraphQL

**Issue link**

Resolves #8 
Depends on #80

### CODE REVIEW

File: installation/resources/cluster-certificate-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
```
Comment (pssap) [inline installation/resources/cluster-certificate-secret.yaml]: This should be ignored by Git.

File: installation/resources/ui-test-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
```
Comment (pssap) [inline installation/resources/ui-test-secret.yaml]: This should be ignored by Git.

Comment (damianpacierpnikatsap) [reply installation/resources/ui-test-secret.yaml]: done

File: installation/resources/cluster-certificate-secret.yaml (hunk 1 above)
Comment (damianpacierpnikatsap) [reply installation/resources/cluster-certificate-secret.yaml]: done

**Comment by damianpacierpnikatsap** [comment]
I will take a look on tests. GraphQL console will not be available without ID token anymore (in fact it was just temporary available without ID token, since we removed out custom adapter for Istio mixer with JWT validation)

**Pull request merged**